ampd/src/mvx/verifier.rs

```diff
@@ -0,0 +1,630 @@
+use crate::handlers::mvx_verify_msg::Message;
+use crate::handlers::mvx_verify_verifier_set::VerifierSetConfirmation;
+use crate::types::Hash;
+use axelar_wasm_std::voting::Vote;
+use base64::engine::general_purpose::STANDARD;
+use base64::Engine;
+use multisig_prover::encoding::mvx::WeightedSigners;
+use multiversx_sdk::data::address::Address;
+use multiversx_sdk::data::transaction::{Events, TransactionOnNetwork};
+
+const CONTRACT_CALL_IDENTIFIER: &str = "callContract";
+const CONTRACT_CALL_EVENT: &str = "contract_call_event";
+
+const ROTATE_SIGNERS_IDENTIFIER: &str = "rotateSigners";
+const SIGNERS_ROTATED_EVENT: &str = "signers_rotated_event";
+
+macro_rules! unwrap_or_continue {
+    ( $data:expr ) => {
+        match $data {
+            Some(x) => x,
+            None => continue,
+        }
+    };
+}
+
+macro_rules! unwrap_or_false {
+    ( $data:expr ) => {
+        match $data {
+            Some(x) => x,
+            None => return false,
+        }
+    };
+}
+
+macro_rules! unwrap_err_or_false {
+    ( $err:expr ) => {
+        match $err {
+            Ok(x) => x,
+            Err(_) => return false,
+        }
+    };
+}
+
+impl PartialEq<&Message> for &Events {
+    fn eq(&self, msg: &&Message) -> bool {

```

Comment:
Changed this to not use the PartialEq and have a function which returns `Result`.
I don't necessarily think that this way is better than the macros, since the errors are just silently ignored, since they are expected to happen and we can return `false` in that case and don't care about what error occurred.